flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/PartitionInfo.java

```diff
@@ -48,4 +54,36 @@ public IntermediateDataSetID getIntermediateDataSetID() {
 	public InputChannelDeploymentDescriptor getInputChannelDeploymentDescriptor() {
 		return inputChannelDeploymentDescriptor;
 	}
+
+	// ------------------------------------------------------------------------
+
+	static PartitionInfo fromEdge(ExecutionEdge executionEdge) {
```

Comment:
`InputChannelDeploymentDescriptor#fromEdges` does something very similar. I think we could refactor `ICDD#fromEdges` to call a `ICDD#fromEdge` and reuse this method here. That way we would get some test coverage for free and avoid code duplication.

Reply:
Yes, I would further refactor ICDD for deduplication.